modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/js/components/PublicationTimeline.js

```diff
@@ -14,119 +14,63 @@ import WorkflowStatusLabel from './WorkflowStatusLabel';
-				publication.ownerName = itemXml.match(
-					/<ownerName>(.*?)<\/ownerName>/
-				)[1];
-				publication.externalReferenceCode = itemXml.match(
-					/<externalReferenceCode>(.*?)<\/externalReferenceCode>/
-				)[1];
-				publication.dateCreated = itemXml.match(
-					/<dateCreated>(.*?)<\/dateCreated>/
-				)[1];
-				publication.dateModified = itemXml.match(
-					/<dateModified>(.*?)<\/dateModified>/
-				)[1];
-
-				publication.status = {};
-
-				publication.status.code = itemXml.match(
-					/<status>.*?<code>(.*?)<\/code>.*?<\/status>/
-				)[1];
-				publication.status.label = itemXml.match(
-					/<status>.*?<label>(.*?)<\/label>.*?<\/status>/
-				)[1];
-				publication.status.label_i18n = itemXml.match(
-					/<status>.*?<label_i18n>(.*?)<\/label_i18n>.*?<\/status>/
-				)[1];
-
-				publication.actions = {};
-
-				publication.actions.schedule = {};
-
-				publication.actions.schedule.method = itemXml.match(
-					/<actions>.*?<schedule>.*?<method>(.*?)<\/method>.*?<\/schedule>.*?<\/actions>/
-				)[1];
-				publication.actions.schedule.href = itemXml.match(
-					/<actions>.*?<schedule>.*?<href>(.*?)<\/href>.*?<\/schedule>.*?<\/actions>/
-				)[1];
-
-				publication.actions.get = {};
-
-				publication.actions.get.method = itemXml.match(
-					/<actions>.*?<get>.*?<method>(.*?)<\/method>.*?<\/get>.*?<\/actions>/
-				)[1];
-				publication.actions.get.href = itemXml.match(
-					/<actions>.*?<get>.*?<href>(.*?)<\/href>.*?<\/get>.*?<\/actions>/
-				)[1];
-
-				publication.actions.publish = {};
-
-				publication.actions.publish.method = itemXml.match(
-					/<actions>.*?<publish>.*?<method>(.*?)<\/method>.*?<\/publish>.*?<\/actions>/
-				)[1];
-				publication.actions.publish.href = itemXml.match(
-					/<actions>.*?<publish>.*?<href>(.*?)<\/href>.*?<\/publish>.*?<\/actions>/
-				)[1];
-
-				publication.actions.update = {};
-
-				publication.actions.update.method = itemXml.match(
-					/<actions>.*?<update>.*?<method>(.*?)<\/method>.*?<\/update>.*?<\/actions>/
-				)[1];
-				publication.actions.update.href = itemXml.match(
-					/<actions>.*?<update>.*?<href>(.*?)<\/href>.*?<\/update>.*?<\/actions>/
-				)[1];
-
-				publication.actions.delete = {};
-
-				publication.actions.delete.method = itemXml.match(
-					/<actions>.*?<delete>.*?<method>(.*?)<\/method>.*?<\/delete>.*?<\/actions>/
-				)[1];
-				publication.actions.delete.href = itemXml.match(
-					/<actions>.*?<delete>.*?<href>(.*?)<\/href>.*?<\/delete>.*?<\/actions>/
-				)[1];
-
-				jsonObject.items.push(publication);
-
-				currentItemStartPos = itemEndPos;
-				itemEndPos = xml.indexOf('</items>', itemEndPos + 1);
+	const createMVCRenderCommandURL = (
+		ctCollectionId,
+		mvcRenderCommandName,
+		additionalParams = {}
+	) => {
+		return createPortletURL(
+			themeDisplay.getLayoutRelativeControlPanelURL(),
+			{
+				ctCollectionId,
+				mvcRenderCommandName,
+				p_p_id:
+					'com_liferay_change_tracking_web_portlet_PublicationsPortlet',
```

Comment:
Let's try to avoid hard-coding the portletId. We can get it from the `namespace` attribute from the ChangeTrackingIndicator component